### PR TITLE
[0262] Vehicle Data Seat Occupancy

### LINKED ISSUE #1240: [SDL 0262] New vehicle data SeatOccupancy
Proposal: [New vehicle data SeatOccupancy](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0262-New-vehicle-data-SeatOccupancy.md)

> This proposal is to add new vehicle data `SeatOccupancy` to know if a vehicle seat is occupied and belted.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/870

Steering Committee Decision:

> The Steering Committee voted to accept this proposal with the following revision: update `SeatOccupancy` parameter description to "Seat status array containing the location and whether the seats are belted."

> The proposal .md file was updated to reflect these revisions on 12/4/19.

### PR DESCRIPTION
**Pull request by kostyaBoss**
Fixes #1240

This PR is **[ready]** for review.

### Related PR's
[Core](https://github.com/smartdevicelink/sdl_core/pull/3585)
[HMI](https://github.com/smartdevicelink/sdl_hmi/pull/467)

### Risk
This PR makes **[minor]** API changes.

### Testing Plan
- [x] I have verified that I have not introduced new warnings in this PR (or explain why below)
- [x] I have run the unit tests with this PR
- [x] I have tested this PR against Core and verified behavior (if applicable, if not applicable, explain why below).
- [x] I have tested Android, Java SE, and Java EE

#### Unit Tests
Added unit tests to cover new structures and update existing

### Summary
Implement changes according to #1240

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by kostyaBoss** [comment]
@santhanamk This pr is ready for Ford review. I do know that CI has failed, though, it seems the problem is not in pr-related features

**Comment by joeygrover** [comment]
@kostyaBoss

1. Please avoid removing sections from the PR template and instead filling them out with detailed and helpful information.
2. You removed and therefore did not check that you have signed the CLA.
3. You commented that this is ready for review, but the description in the PR states it is not.
4. You do not provide a Core branch/PR that this was tested against and therefore this PR is not reviewable at this time.
5. Multiple files are completely overwritten, this makes a code review very difficult and is in almost all cases incorrect. It is likely you simply ran the RPC generator and committed the results. This is not an acceptable method for the Java Suite library as there are many corner cases in the RPC classes that the generator does not have information on. The generator is used for new classes, however, modification on existing classes should not come from the generator as it will overwrite the file as seen in this PR. The committed changes should only be the code that actually changed.

**Comment by vladmu** [comment]
@joeygrover thank you for your feedback, we appreciate your help and will fix all problems you mentioned. However please note that this PR currently is under internal and Ford review, therefore some problems could be present here. We'll do all our best to minimize them before this PR goes to the PMs review.

**Comment by joeygrover** [comment]
@vladmu I am not performing a content review of the code itself, but as the project maintainer if I see a pull request that is very off it is in the best interest of the SDLC to comment on it right away and ensure that those items can be addressed as to avoid delays. Any pull request in the open will always be visible to the community and will be open to comments, especially those from the project maintainer. We await Ford's approval before performing a thorough review.

File: android/sdl_android/src/androidTest/java/com/smartdevicelink/test/TestValues.java
Comment (santhanamk) [inline android/sdl_android/src/androidTest/java/com/smartdevicelink/test/TestValues.java:435]: @kostyaBoss I would move lines 435 and 437 (`JSONArray` objects) to below line 563 where the other `JSONArray` declarations are.

File: android/sdl_android/src/androidTest/java/com/smartdevicelink/test/TestValues.java
Comment (santhanamk) [inline android/sdl_android/src/androidTest/java/com/smartdevicelink/test/TestValues.java:434]: @kostyaBoss I would add this declaration `public static final JSONObject JSON_OBJECT_GENERAL_SEATS_OCCUPIED = new JSONObject();` where the other `JSONObjects` are declared (around line 604)

File: android/sdl_android/src/androidTest/java/com/smartdevicelink/test/TestValues.java
Comment (santhanamk) [inline android/sdl_android/src/androidTest/java/com/smartdevicelink/test/TestValues.java:435]: @kostyaBoss I would add this declaration `public static final JSONObject JSON_OBJECT_GENERAL_SEATS_BELTED = new JSONObject();` where the other `JSONObjects` are declared (around line 604)

File: android/sdl_android/src/androidTest/java/com/smartdevicelink/test/TestValues.java
Comment (santhanamk) [inline android/sdl_android/src/androidTest/java/com/smartdevicelink/test/TestValues.java:1179]: @kostyaBoss I would delete lines 1179-1189.

**Review by santhanamk — COMMENTED** [review]
@kostyaBoss Code looks good.  I have suggested a few changes.  Please review, and make the changes as needed.

**Review by santhanamk — COMMENTED** [review]
I would replace lines 1179-1189 with 

`            JSON_OBJECT_GENERAL_SEATS_OCCUPIED.put(SeatStatus.KEY_SEAT_LOCATION, GENERAL_SEAT_LOCATION);
            JSON_OBJECT_GENERAL_SEATS_OCCUPIED.put(SeatStatus.KEY_CONDITION_ACTIVE,GENERAL_BOOLEAN);
            JSON_GENERAL_SEATS_OCCUPIED.put(JSON_OBJECT_GENERAL_SEATS_OCCUPIED);
            JSON_OBJECT_GENERAL_SEATS_BELTED.put(SeatStatus.KEY_SEAT_LOCATION, GENERAL_SEAT_LOCATION);
            JSON_OBJECT_GENERAL_SEATS_BELTED.put(SeatStatus.KEY_CONDITION_ACTIVE,GENERAL_BOOLEAN);
            JSON_GENERAL_SEATS_BELTED.put(JSON_OBJECT_GENERAL_SEATS_BELTED);`

File: android/sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/datatypes/SeatStatusTests.java
Comment (kostyaBoss) [reply android/sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/datatypes/SeatStatusTests.java:4]: @santanamk, Grid import is used at the line 67

**Comment by kostyaBoss** [comment]
@santhanamk I've updated the PR please re-review

File: base/src/main/java/com/smartdevicelink/proxy/rpc/GetVehicleData.java
Comment (santhanamk) [inline base/src/main/java/com/smartdevicelink/proxy/rpc/GetVehicleData.java]: @kostyaBoss It looks like the entire file `GetVehicleData.java` is being regenerated by the code generator tool.  Can you just merge in the changes for this PR (data related to `seatOccupancy`)?

File: base/src/main/java/com/smartdevicelink/proxy/rpc/GetVehicleDataResponse.java
Comment (santhanamk) [inline base/src/main/java/com/smartdevicelink/proxy/rpc/GetVehicleDataResponse.java]: @kostyaBoss It looks like the entire file `GetVehicleDataResponse.java` is being regenerated by the code generator tool.  Can you just merge in the changes for this PR (data related to `seatOccupancy`)?

File: base/src/main/java/com/smartdevicelink/proxy/rpc/OnVehicleData.java
Comment (santhanamk) [inline base/src/main/java/com/smartdevicelink/proxy/rpc/OnVehicleData.java]: @kostyaBoss It looks like the entire file `OnVehicleData.java` is being regenerated by the code generator tool.  Can you just merge in the changes for this PR (data related to `seatOccupancy`)?

File: android/sdl_android/src/androidTest/java/com/smartdevicelink/test/Validator.java
Comment (santhanamk) [inline android/sdl_android/src/androidTest/java/com/smartdevicelink/test/Validator.java:3924]: @kostyaBoss  On line 3924 I would rename `seatStatuses2` to `seatsBelted2`.

Comment (santhanamk) [inline android/sdl_android/src/androidTest/java/com/smartdevicelink/test/Validator.java:3924]: @kostyaBoss On line 3921 I would rename `seatStatuses1` to `seatsBelted1`.

File: android/sdl_android/src/androidTest/java/com/smartdevicelink/test/VehicleDataHelper.java
Comment (santhanamk) [inline android/sdl_android/src/androidTest/java/com/smartdevicelink/test/VehicleDataHelper.java:98]: @kostyaBoss Line 97 is in the right place, but you can move the other variables on lines 98-101 back to where they were before.  

Line 97 `SEAT_OCCUPANCY` is the top level variable while the other variables on lines 98-101 are lower variables.

**Review by santhanamk — COMMENTED** [review]
@kostyaBoss I reviewed the changes made from the code review.  It looks good.  I have  a few more suggestions.  Please take a look whenever you get the chance.

**Comment by kostyaBoss** [comment]
@santhanamk I've updated the PR, please re-review

**Review by santhanamk — APPROVED** [review]
@kostyaBoss Code looks good. I have approved the PR.

Once you have tested against Core and HMI (if applicable), please put the links for those PRs. Please mark the PR as ready for review, and you can tag Livio once the testing is done on your end.

**Comment by kostyaBoss** [comment]
@bilal-alsharifi , @jordynmackool the PR is ready for the review

**Review by bilal-alsharifi — CHANGES_REQUESTED** [review]

File: android/sdl_android/src/androidTest/java/com/smartdevicelink/test/TestValues.java
Comment (kostyaBoss) [reply android/sdl_android/src/androidTest/java/com/smartdevicelink/test/TestValues.java:114]: @bilal-alsharifi I've checked the failing test, it's non-related for this PR test `SeatLocationCapabilityTests`, line 91. Previously we were not changing the tests which are not related to the PR. Please, let me know if this has changed

Comment (bilal-alsharifi) [reply android/sdl_android/src/androidTest/java/com/smartdevicelink/test/TestValues.java:114]: It is passing on develop though so something changed in the PR that made the test fail.

**Comment by kostyaBoss** [comment]
@bilal-alsharifi I've processed the test, please, re-review

**Review by bilal-alsharifi — APPROVED** [review]

**Comment by bilal-alsharifi** [comment]
@kostyaBoss It seems that there are merge conflicts now. Can you pull in develop to fix the conflicts?

**Pull request merged**
